### ISSUE TITLE
Not able to handle pages in subdirectories

### ISSUE THREAD

**Issue by toniocus** [andriusvelykis/reflow-maven-skin#13]
I was trying to use reflow-maven-skin with an already done site, and face an important problem:

I have my pages subdivided in this directory structure

```
apt/index.apt
apt/dev/developer01.apt
apt/dev/developer02.apt
......
apt/user/user01.apt
apt/user/user02.apt
......
```

This is working correctly in standard maven-site skin when calling, or with fluido skin.

```
mvn site:run
```

But when using reflow-maven-skin once I get into any page inside
**user** or **dev** directory I cannot reach any other page.

PAGE NOT FOUND Error is returned

So for now just keep using old skins.

**Comment by andriusvelykis** [comment]
Hi - sorry for the delay. Would you mind attaching a minimal example demonstrating this? It would be easier to check.

**Comment by gerhardgossen** [comment]
I ran into the same problem and created a minimal example at https://github.com/gerhardgossen/reflow-13. 

To reproduce:
1. run `mvn site` on this project
2. open the generate site in your browser
3. click on "dev" in the bottom menu
4. click on "user" in the bottom menu, which leads to a "file not found"-error, because this link leads to `.../site/dev/user/index.html` instead of `.../site/user/index.html`.

Using the default skin the link is created correctly.

This problem also occurs for images (e.g. the banner image), they do not appear in files in subdirectories.

**Comment by andriusvelykis** [comment]
Thanks for the example - that did help to show the issue. I have fixed this in b289d676c0b1fbea6d5bbbac77c9870df6036419.

The problem was arising from extra normalisation of relative links that Reflow skin is doing. It is needed for 'active' link calculation, e.g. to highlight links to the current page. Other skins do not seem to be doing that. In your example, when you open /dev/index.html, you can see that the "Dev" link is actually `../dev/index.html`, rather than empty or `index.html`. Reflow skin performs additional normalisation to get the empty link and thus indicate it as 'active'.

The problem was that to do that, the skin expected `project.url` to be set. This also helps with highlighting absolute links as 'active'.. I have fixed it so that if no `project.url` is set in `pom.xml`, I use a dummy URL to relativize the links.

Thanks to you both for reporting - I have pushed the updated SNAPSHOT build. Let me know if you still encounter the problem.

**Comment by gerhardgossen** [comment]
Works both for the minimal example as well as for my actual project. Thanks for the quick turnaround!
